pkg/apis/pipeline/v1alpha1/taskrun_types.go

```diff
@@ -125,10 +124,9 @@ type TaskRun struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// +optional
-	Spec TaskRunSpec `json:"spec,omitempty"`
+	Spec *TaskRunSpec `json:"spec,omitempty"`
```

Comment:
why are we changing these to pointers?

Reply:
These became pointers b/c json marshalling will cause fields to be required, even if they are marked optional, unless they are nullable (https://stackoverflow.com/questions/18088294/how-to-not-marshal-an-empty-struct-into-json-with-go)

(Let me know if you know of a better way to handle this tho!)

Actually though, now that I think about it I don't know why the spec would be optional, so maybe I'll undo this change and make this required instead :thinking: